### ISSUE TITLE
Reproducing Cascade Mask R-CNN on caltech dataset 

### ISSUE THREAD

**Issue by Jonghu** [hasanirtiza/Pedestron#121]
First of all, thank you for sharing this wonderful repository and good paper!

I'm afraid I have trouble with reproducing MR result of caltech dataset with the pretrained model you've provided at [README](https://github.com/hasanirtiza/Pedestron#benchmarking-of-pre-trained-models-on-pedestrian-detection-datasets-autonomous-driving).

I have three questions

# 1. Reproduction MR discrepancy

With pretrained weight "Cascade Mask R-CNN / Caltech / HRNet / 1.7 /25.7" `epoch_14.pth.stu`
- Claimed MR: 1.7
- My run: 1.504878 
The result is **better** than that of yours in README which is super mysterious. Are there things I might have missed?

# 2. Unexpected key in the pretrained weight
While loading pretrained model I got the error, though the script ran without extra error raise.
```
unexpected key in source state_dict: mask_head.0.conv_res.conv.weight, mask_head.0.conv_res.conv.bias, mask_head.1.conv_res.conv.weight, mask_head.1.conv_res.conv.bias, mask_head.2.conv_res.conv.weight, mask_head.2.conv_res.conv.bias
```
I used `configs/elephant/caltech/cascade_hrnet.py` as config file.
I could've found out that changing `configs/elephant/caltech/cascade_hrnet.py#L123` to 'HTCMaskHead' would fix 'unexpected keys warning', though the model forward pass wouldn't work.

Is this as expected? Could it be related to the problem 1?

# 3. The paper's result on caltech dataset
**I couldn't find MR of 1.7 at the paper "Generalizable Pedestrian Detection: The Elephant In The Room" regarding caltech dataset.** 
The best MR of caltech that I could've found in the paper was **2.5 at <Table 8>**. 
Could you let me know where the number 1.7 is?

# How I tried
1. Clone latest code (commit id 8412ee27ba8011b9891df8c239b6cf644d69bb82)
2. Followed docker installation as provided ([Link](https://github.com/hasanirtiza/Pedestron/blob/master/INSTALL.md#docker-installation))
3. Followed dataset preparation as provided ([Link](https://github.com/hasanirtiza/Pedestron/blob/master/INSTALL.md#docker-installation))
    1. Download caltech videos from [link](http://www.vision.caltech.edu/Image_Datasets/CaltechPedestrians/)
    2. Export video frames as images with below directory structure.
 ```
datasets/Caltech/
|__train_images/  # 128419 images
    |__set00_V000_I00000.jpg
    |__set00_V000_I00001.jpg
    |__ ...
    |__set05_V012_I01706.jpg
|__test_images # 121465 images
    |__set06_V000_I00000.jpg
    |__set06_V000_I00001.jpg
    |__ ...
    |__set10_V011_I01733.jpg
```
4. Downloaded weight "Cascade Mask R-CNN / Caltech / HRNet / 1.7 /25.7" `epoch_14.pth.stu` [google drive](https://drive.google.com/file/d/12J5z4G48LIgv2w74VvH6jz-Z2095SIDU/view) 
4. Make predictions: `python tools/test_caltech.py configs/elephant/caltech/cascade_hrnet.py epoch_ 14 15 --out result_caltech --mean_teacher`
5. Output conversion: `python tools/caltech/convert_json_to_txt.py`
     1.  Change`tools/caltech/convert_json_to_txt.py#L9` `result_dir='.'`
     2. Move txt files to be place at proper place.
```
eval_caltech/
|__ Pedestron_Result/
   |__result_caltech/
       |__set06/
           |__V000.txt
           |__...
       |__set07 ... set10
|__ResultsEval/
   |__ ... (As provided in the repo)
|__dbEval.m
|__ ... (and other matlab scripts)
 ```
6. Matlab evaluation: Run `dbEval.m`

**Thank you for your precious time!**

**Comment by hasanirtiza** [comment]
Hard to say that why are you getting better results (I did remember post submission we had a better trained model, so potentially there is a chance that it is a typo), just to double check, are you using the `gt file` we provided in the repo ?

**Comment by Jonghu** [comment]
Thanks for the reply!

As far as I understood, I used `tools/caltech/eval_caltech/ResultsEval/gt-new/gt-Reasonable.mat` [link](https://github.com/hasanirtiza/Pedestron/tree/master/tools/caltech/eval_caltech/ResultsEval/gt-new)
which is default setting of `dbEval.m` (I haven't touched anything in the script.)

**Comment by Jonghu** [comment]
@hasanirtiza 
Sorry for bothering you, but I'll make the questions more clear!

1. What setting do I need to reproduce "Cascade Mask R-CNN / Caltech / HRNet / 1.7 / 25.7" at [the table](https://github.com/hasanirtiza/Pedestron#benchmarking-of-pre-trained-models-on-pedestrian-detection-datasets-autonomous-driving)?
Especially for the **training set**, **test set**, config file, etc.
2. Which table and row in the paper does directly related to the above weight? (Cascade Mask R-CNN / Caltech / HRNet / 1.7 / 25.7) Or is it not related to the paper's result?
I totally understood the situation of post submission. I just want to check what exact experiment setting the result came from.

Thanks in advance!

**Comment by hasanirtiza** [comment]
1) As we write in the Table 8 of our CVPR paper, you need the progressive fine tuning strategy. Pre-train it on WiderPedestrian  and subsequently fine tune on ECP (you can evaluate on Caltech after this step as well, you should get results in the same ball park) and finally do another round of fine tuning on Caltech (usually one or two epochs are enough).   

2) None, not related to the CVPR results. These results are not included in our submission as of now, but they are obtained through progressive fine-tuning. We are working on an upcoming submission, they will be included in there. It is not public yet. 

Hope that helps.

**Comment by Jonghu** [comment]
Thank you so much for the clear explanation.
It helped me a lot.
